File: src/main/java/gg/arcdev/practice/game/arena/impl/SharedArena.java
package gg.arcdev.practice.game.arena.impl;

import gg.arcdev.practice.game.arena.Arena;
import gg.arcdev.practice.game.arena.ArenaType;
import gg.arcdev.practice.util.LocationUtil;
import java.io.IOException;
import lombok.Getter;
import lombok.Setter;
import gg.arcdev.practice.Main;
import org.bukkit.Location;
import org.bukkit.configuration.file.FileConfiguration;

@Getter
@Setter
public class SharedArena extends Arena {

	public SharedArena(String name, Location location1, Location location2) {
		super(name, location1, location2);
	}

	@Override
	public ArenaType getType() {
		return ArenaType.SHARED;
	}

	@Override
	public void save() {
		String path = "arenas." + getName();

		FileConfiguration configuration = Main.getInstance().getArenasConfig().getConfiguration();
		configuration.set(path, null);
		configuration.set(path + ".type", getType().name());
		configuration.set(path + ".spawnA", LocationUtil.serialize(spawnA));
		configuration.set(path + ".spawnB", LocationUtil.serialize(spawnB));
		configuration.set(path + ".cuboid.location1", LocationUtil.serialize(getLowerCorner()));
		configuration.set(path + ".cuboid.location2", LocationUtil.serialize(getUpperCorner()));
		configuration.set(path + ".kits", getKits());

		try {
			configuration.save(Main.getInstance().getArenasConfig().getFile());
		} catch (IOException e) {
			e.printStackTrace();
		}
	}

	@Override
	public void delete() {
		super.delete();

		FileConfiguration configuration = Main.getInstance().getArenasConfig().getConfiguration();
		configuration.set("arenas." + getName(), null);

		try {
			configuration.save(Main.getInstance().getArenasConfig().getFile());
		} catch (IOException e) {
			e.printStackTrace();
		}
	}

}

File: src/main/java/gg/arcdev/practice/game/arena/impl/SharedArena.kt
package gg.arcdev.practice.game.arena.impl

import gg.arcdev.practice.Main
import gg.arcdev.practice.game.arena.Arena
import gg.arcdev.practice.game.arena.ArenaType
import gg.arcdev.practice.util.LocationUtil
import org.bukkit.Location
import java.io.IOException

class SharedArena(name: String, location1: Location, location2: Location) : Arena(name, location1, location2) {

    override fun getType(): ArenaType = ArenaType.SHARED

    override fun save() {
        val path = "arenas.$name"
        val configuration = Main.getInstance().getArenasConfig().configuration

        configuration.set(path, null)
        configuration.set("$path.type", getType().name)
        configuration.set("$path.spawnA", LocationUtil.serialize(spawnA))
        configuration.set("$path.spawnB", LocationUtil.serialize(spawnB))
        configuration.set("$path.cuboid.location1", LocationUtil.serialize(lowerCorner))
        configuration.set("$path.cuboid.location2", LocationUtil.serialize(upperCorner))
        configuration.set("$path.kits", kits)

        try {
            configuration.save(Main.getInstance().getArenasConfig().file)
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }

    override fun delete() {
        super.delete()

        val configuration = Main.getInstance().getArenasConfig().configuration
        configuration.set("arenas.$name", null)

        try {
            configuration.save(Main.getInstance().getArenasConfig().file)
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }
}
